### PR TITLE
[feature](Nereids) eliminate sort under subquery

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

eliminate sort under subquery

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/EliminateSortTest.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 /**
  * column prune ut.
  */
-public class EliminateSortTest extends TestWithFeService implements MemoPatternMatchSupported {
+class EliminateSortTest extends TestWithFeService implements MemoPatternMatchSupported {
```
Comment (xzj7019) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/EliminateSortTest.java:29]: better to add ut to cover "select * from (select xxx order by xxx) a left join (select xxx ) b on a.id =b.id limit n;" also.

**Comment by XieJiann** [comment]
run buildall

**Review by xzj7019 — APPROVED** [review]

**Comment by XieJiann** [comment]
run buildall

**Review by jackwener — APPROVED** [review]

**Pull request merged**